doc/OnlineDocs/principles.rst

```diff
@@ -0,0 +1,193 @@
+  (e.g., ``sphinx``, ``sphinx_rtd_theme``, ``numpydoc``).
+
+* **Optional** – Dependencies that enable extended
+  functionality throughout the Pyomo codebase, such as numerical
+  computations or data handling (e.g., ``numpy``, ``pandas``,
+  ``matplotlib``).
+
+These optional dependencies can be installed selectively using standard
+``pip`` commands. For example:
+
+::
+
+   pip install pyomo[optional]
+   pip install pyomo[tests,docs]
+
+Pyomo's continuous integration infrastructure regularly tests against
+the most recent versions of all optional dependencies to ensure that
+Pyomo remains compatible with current releases in the Python ecosystem.
+When incompatibilities are identified, the setup configuration is
+updated accordingly.
+
+.. note::
+
+   For more information on installing Pyomo with optional extras,
+   see :doc:`/getting_started/installation`.
+
+Solvers
++++++++
+
+Pyomo does not bundle or directly distribute optimization solvers.
+We recognize that solver installation can be challenging for new users.
+To assist with this process, see the solver
+availability table and installation guidance in
+:doc:`/getting_started/solvers`. This table lists solvers that can be
+installed via ``pip`` or ``conda`` where available, but Pyomo itself
+does not include or require any specific solver as a dependency.
+
+Miscellaneous Conventions
+-------------------------
+
+There are a variety of long-standing conventions that have become
+standard across the project. This list will be amended as conventions come
+up, so please refer regularly for updates:
+
+* **Environment imports:** Import the main Pyomo environment as  
+  ``import pyomo.environ as pyo``.  
+  Avoid ``from pyomo.environ import *`` or alternative aliases.
+  Additionally, avoid all uses of ``import *``.
+
+* **Export lists:** Do not define ``__all__`` in modules.  
+  Public symbols are determined by naming and documentation, not explicit lists.
+
+* **Circular imports:** Avoid importing from ``pyomo.core`` into any module
+  in ``pyomo.common`` due to the potential for circular imports.
+
+* **Print statements:** Avoid printing directly to ``stdout``. Pyomo is a
+  library, not an application, and copious output can interfere with downstream
+  tools and workflows. Use the appropriate logger instead. Print
+  information only when the user has enabled or requested it.
+
+* **Pull Request naming:** Pull Request titles are added to the CHANGELOG
+  and the release notes. The Pyomo development team reserves the right to
+  alter titles as appropriate to ensure they fit the look and feel of
+  other titles in the CHANGELOG.
+
+* **Full commit history:** We do **not** squash-merge Pull Requests,
+  preferring to retain the entire commit history.
+
+* **URLs:** All links in code, comments, and documentation must use ``https`` 
+  rather than ``http``.
+
+* **File headers:** Every ``.py`` file must begin with the standard Pyomo
+  copyright header:
+
+  .. code-block:: text
```

Comment:
Can this be `.. literalinclude::`'ed from `pyomo/__init__.py` (that way the documentation is kept in sync with the codebase)?